ts/features/bonus/bpd/screens/optInPaymentMethods/OptInPaymentMethodsCashbackUpdateScreen.tsx

```diff
@@ -1,36 +1,134 @@
-import { View } from "native-base";
+import {
+  Image,
+  ImageSourcePropType,
+  SafeAreaView,
+  ScrollView,
+  StyleSheet
+} from "react-native";
+import { View, Text } from "native-base";
 import I18n from "../../../../../i18n";
 import BaseScreenComponent from "../../../../../components/screens/BaseScreenComponent";
 import { IOStyles } from "../../../../../components/core/variables/IOStyles";
-import { H1 } from "../../../../../components/core/typography/H1";
 import ButtonDefaultOpacity from "../../../../../components/ButtonDefaultOpacity";
+import { IORenderHtml } from "../../../../../components/core/IORenderHtml";
+import { H2 } from "../../../../../components/core/typography/H2";
+import { useIODispatch, useIOSelector } from "../../../../../store/hooks";
+import { availableBonusTypesSelectorFromId } from "../../../bonusVacanze/store/reducers/availableBonusesTypes";
+import { ID_BPD_TYPE } from "../../../bonusVacanze/utils/bonus";
+import dafaultLogo from "../../../../../../img/bonus/bpd/logo_cashback_blue.png";
+import FooterWithButtons from "../../../../../components/ui/FooterWithButtons";
+import { confirmButtonProps } from "../../../bonusVacanze/components/buttons/ButtonConfigurations";
+import { navigateToOptInPaymentMethodsChoiceScreen } from "../../navigation/actions";
+import { useNavigationContext } from "../../../../../utils/hooks/useOnFocus";
+import {
+  getBpdAMethodsSelector,
+  paymentMethodsSelector
+} from "../../../../../store/reducers/wallet/wallets";
+import {
+  optInPaymentMethodsCompleted,
+  optInPaymentMethodsFailure
+} from "../../store/actions/optInPaymentMethods";
 
-const OptInPaymentMethodsCashbackUpdateScreen = () => (
-  // The void customRightIcon and customGoBack are needed to have a centered header title
-  <BaseScreenComponent
-    showInstabugChat={false}
-    goBack={false}
-    headerTitle={I18n.t("bonus.bpd.optInPaymentMethods.cashbackUpdate.header")}
-    customRightIcon={{
-      iconName: "",
-      onPress: () => true
-    }}
-    customGoBack={
-      <ButtonDefaultOpacity onPress={() => true} transparent={true} />
+const styles = StyleSheet.create({
+  logo: {
+    resizeMode: "contain",
+    width: 48,
+    height: 46
+  },
+  headerContainer: {
+    ...IOStyles.row,
+    justifyContent: "space-between"
+  }
+});
+const OptInPaymentMethodsCashbackUpdateScreen = () => {
+  const navigation = useNavigationContext();
+  const dispatch = useIODispatch();
+  const bpdPaymentMethods = useIOSelector(getBpdAMethodsSelector);
+  const paymentMethods = useIOSelector(paymentMethodsSelector);
+  const bpdInfo = useIOSelector(availableBonusTypesSelectorFromId(ID_BPD_TYPE));
+  const bpdLogo: ImageSourcePropType = bpdInfo?.cover
+    ? { uri: bpdInfo?.cover }
+    : dafaultLogo;
+
+  // This screen should be shown only if the payment method are correctly loaded
+  if (paymentMethods.kind !== "PotSome") {
+    dispatch(optInPaymentMethodsFailure("error in the payment method loading"));
+  }
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❔ question:** ‏Just a question to see if I understand correctly the behaviour, even in the case of refresh of payment methods (`someLoading`) the user have to see the failure screen, is this right?